raster-foundry/raster-foundry
# Add first cut permissions modal

## Overview

This PR adds a barebones, ugly, unstyled permissions modal for adding permissions to first-class objects. It handles:

- [x] adding a new permission
- [x] updating permissions that are already defined
- [x] removing permissions
- [x] additional action types depending on context
- [x] sharing with a user's organization
- [x] sharing with teams a user is a member of
- [x] sharing with other users in a user's platform (blocked by #3452 but it should work)
- [x] sharing with everyone

### Checklist


### Notes

It's ugly I know I know there are classes in place to target for styling I'm sorry it's so ugly right now.

More importantly:

it doesn't really respect the designs we have, for two reasons. The first is that we didn't know they
existed when I started. The second is that, since our permissions are non-hierarchical and require
support for context-sensitive types of actions (annotating projects, downloading scenes...), the
single permission dropdown didn't really make sense as a strategy for creating grants. There's 
more work to be done here to work out how we're going to get everything tied together, but
this is a first cut.

Also I'm not sure about the `CONFIRM` workflow -- I think probably it's worth checking, since I think gmail makes you confirm that you intend to give someone permissions, and that's our inspiration to
some extent. But also it seems maybe cumbersome? Like I don't know if that's going to annoy people
or make them think that we're taking their data security seriously.

I'll rebase out the dumb button in the raster list before merging once we figure out how to get into
this context.

## Testing Instructions

 * make some teams and add your dev user to them ([these instructions](https://github.com/azavea/raster-foundry-platform/wiki/Managing-platforms-and-user-roles) may be helpful) 
 * go to your raster list area
 * click on the red download button (I just reused the icon because it's probably going away anyway)
 * add a new permission for teams or orgs
 * confirm that you want to add them
 * examine your new permissions table and confirm it respected your wishes
 * add more permissions and confirm results are what you'd expect

Closes #3384
